### ISSUE TITLE
Can't add new plugins to Kibana

### ISSUE THREAD

**Issue by spaceCamel** [NixOS/nixpkgs#16795]
## Issue description

I want to install "elastic/sense" plugin in kibana, but I can't find a way to do it.

Looking into `kibana.nix` I noticed there is a static `bundled_plugin_ids` list. I'm not sure if "elastic/sense" should be added there and how. 

I tried to use kibana command line to install the plugin as shown below, but that failed too because the plugin directory is read-only.
### Steps to reproduce

Add following services to `/etc/nixos/configuration.nix` and switch:

```
services {
    elasticsearch = {
        enable = true;
        dataDir = "/home/elasticsearch";
    };
    kibana {
        enable = true;
        dataDir = "/home/kibana";
    };
}
```

Run from CLI:

```
# sudo -u kibana BABEL_CACHE_PATH=/home/kibana/.babelcache.json kibana plugin -i elastic/sense
Installing sense
Plugin installation was unsuccessful due to error "EROFS: read-only file system, mkdir '/nix/store/b7fd6i35ym8axl0v7ip9sfhai9grn8ki-kibana-4.4.1/libexec/kibana/installedPlugins/.plugin.installing'"
```

Note, I configured `services.kibana.dataDir = "/home/kibana"` so I had to pass `BABEL_CACHE_PATH` on the CLI otherwise Kibana would try and fail writing the ".babelcache.json" file in "/nix/store/..."
## Technical details
- System: `16.03.1028.0546a4a (Emu)`
- Nix version: `nix-env (Nix) 1.11.2`
- Nixpkgs version: `16.03.1028.0546a4a`

**Comment by 0xABAB** [comment]
There are some other programs in nixpkgs (e.g. for weechat there is a PR), which have implemented plugins, so you could copy from that, if you are still interested that is. If you are not, please mention that instead.

**Comment by GlennS** [comment]
I've built a version of Kibana which includes the [LogTrail](https://github.com/sivasamyk/logtrail) plugin.

I'm putting some information about how I did it here, in case it helps someone.

Plugins go into `$out/libexec/kibana/plugins/$pluginName`, so I modified the `installPhase` of the Kibana package to include:
`ln -s "${logtrail.src}/logtrail" "$out/libexec/kibana/plugins/logtrail"`.

Kibana has an _optimize_ step which minifies CSS and JS. It runs this on first startup (or whenever Kibana's `system.basePath` property changes). This will fail, because the directory it's trying to write to is read only.

I see two ways to get around this:

 1. Run Kibana once as part of the install: `$out/bin/kibana --elasticsearch.requestTimeout 1 || echo "Kibana optimize step complete."`. This is probably the correct thing for Nix, but I wasn't able to convince Kibana not to rebuild these.
 2. Set the `optimize.bundleDir` Kibana config setting to something mutable. I created a `/var/lib/kibana/bundles` directory. This is what I ended up doing.

It looks like future versions of Kibana will do option _(2)_ by default: https://github.com/elastic/kibana/issues/5071 https://github.com/elastic/kibana/pull/25944
